server/api/lib/trees/regional_tree_permit.js

```diff
@@ -0,0 +1,233 @@
+
+	'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/Befor_merkaz_shfela.XLS',
+	'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/after_merkaz_shfela.XLS',
+
+	'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/Befor_jerusalem.XLS',
+	'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/after_jerusalem.XLS',
+
+	'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/Befor_darom.XLS',
+	//'https://www.moag.gov.il/yhidotmisrad/forest_commissioner/rishyonot_krita/Documents/after_darom.XLS', - no after darom
+
+];
+
+const SHEET_BEFORE = 'Data2ToExcel_BeforDate';
+const SHEET_AFTER = 'Data2ToExcel_ToDate';
+const TIMEOUT_MS = 5000;
+
+const bucketName = Config.get('aws.treeBucketName');
+const useS3 = Config.get('aws.useS3ForTreeFiles');
+
+async function getRegionalTreePermitsFromFile(url, pathname) {
+	try {
+		const controller = new AbortController();
+		const controllerTimeout = setTimeout(
+			() => { controller.abort(); },
+			TIMEOUT_MS,
+		);
+		Log.info('Fetching trees file... ' + `${url}`);
+		return new Promise((resolve, reject) => {
+			(async () => {
+				try {
+					const res = await fetch(url, { signal: controller.signal });
+					const stream = fs.createWriteStream(pathname);
+					stream.on('open', () => {
+						res.body.pipe(stream);
+					});
+					stream.on('finish', async function () {
+						stream.close();
+						Log.info(`Sucessfully Downloaded trees file: ${url}
+				 File Could be found here: ${pathname}`);
+
+						const treePermits = await parseTreesXLS(pathname);
+						resolve(treePermits);
+					});
+				}
+				catch (err) {
+					Log.error(`Error fetching file ${url} :  ${err}`);
+					reject(err);
+				}
+				finally {
+					() => { clearTimeout(controllerTimeout); };
+				}
+			})();
+		});
+	}
+	catch (err) {
+		Log.error(err);
+		return Promise.reject();
+	}
+}
+
+async function saveNewTreePermits(treePermits) {
+	// Tree permits are published for objecctions for a period of 2 weeks. taking a 12 months
+	// buffer should be enough for human to remove those lines from the excel sheet.
+	//We're reading a the rows as a bulk and match them at compute time for performance.
+
+	if (treePermits.length == 0) return [];
+	// all tree permits in a chunk should be from the same regional office
+	const regionalOffice = treePermits[0].attributes[tpc.REGIONAL_OFFICE];
+	// this is the only timestamp format knex correcrtly work with 
+	const time_ago = moment().subtract(1, 'year').format('YYYY-MM-DDTHH:mm:ssZ');
+	const existingPermitsCompact = new Set();
+	await database.Knex(tpc.TREE_PERMIT_TABLE).where('updated_at', '>', time_ago.toString())
+		.andWhere(tpc.REGIONAL_OFFICE, regionalOffice)
+		.then(rows => {
+			for (let row of rows) {
+				const key_as_string =
+					[
```

Comment:
Why not use a template string instead of the join ('_')? 
Also, we can use a map function instead of the for loop

existingPermitsCompact.add(rows.map(...))